File: cli-tools/platform/node/list-templates.js
const axios = require('axios');
const { BASE_URL, getAuthParams } = require('./utils');

async function listTemplates() {
    try {
        const params = getAuthParams();
        const response = await axios.get(`${BASE_URL}/Templates`, { params });
        
        if (response.data && response.data.Templates) {
            console.log(JSON.stringify(response.data.Templates, null, 2));
        } else {
            console.log("No templates found or error in response.");
            console.log(response.data);
        }
    } catch (error) {
        console.error("Error fetching templates:", error.message);
    }
}

listTemplates();

File: cli-tools/platform/python/list_templates.py
import requests
import json
from utils import BASE_URL, get_auth_params

def list_templates():
    try:
        params = get_auth_params()
        response = requests.get(f"{BASE_URL}/Templates", params=params)
        response.raise_for_status()
        
        data = response.json()
        if 'Templates' in data:
            print(json.dumps(data['Templates'], indent=2))
        else:
            print("No templates found or error in response.")
            print(data)
            
    except Exception as e:
        print(f"Error fetching templates: {e}", file=sys.stderr)

if __name__ == "__main__":
    list_templates()
